### ISSUE TITLE
Applying CA1853 produces incorrect code that changes program behavior 

### ISSUE THREAD

**Issue by Yozer** [dotnet/roslyn-analyzers#6274]
### Analyzer

**Diagnostic ID**: [CA1853](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1853): `Do not guard 'Dictionary.Remove(key)' with 'Dictionary.ContainsKey(key)'`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet7/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/overview/7.0.0-preview1.22559.1

### Describe the bug

Applying the suggested fix for CA1853 produces code that behaves differently.

### Steps To Reproduce

```csharp
var statuses = new Dictionary<string, string> { { "code", "test" } };
var removedStatuses = new Dictionary<string, string>();

void RemoveStatus(string key, string status)
{
    if (statuses.ContainsKey(key))
    {
        statuses.Remove(key);
        removedStatuses.Add(key, status);
    }
}

RemoveStatus("code", "why not");
Console.WriteLine(removedStatuses.Count);
```

### Expected behavior
No fix being suggested or something this:
```csharp
    if (statuses.Remove(key))
    {
        removedStatuses.Add(key, status);
    }
```
### Actual behavior
Code inside if statement is being removed:
![image](https://user-images.githubusercontent.com/9254709/202248592-4d69860b-5e9b-4323-8a52-df7f9b938ba6.png)
## Additional context

Visual Studio: 17.4
.NET 7
```<AnalysisLevel>latest-Minimum</AnalysisLevel>```

**Comment by mpidash** [comment]
There is another wrong fix applied to the following code:

```C#
Dictionary<int, int> dict = new();

if (!dict.ContainsKey(0))
{
    dict.Remove(0);
}
```

This will suggest a fix and produce the following:

```C#
dict.Remove(0);
```

---

This and the original bug in this issue is a consequence of `SyntaxSupportedByFixer` being wrong, both for the C# and VB implementation:

https://github.com/dotnet/roslyn-analyzers/blob/45879f0b46557886b8d553d121a81866f67ce08a/src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpDoNotGuardDictionaryRemoveByContainsKeyFixer.cs#L19-L23

The check for `ConditionalExpressionSyntax` is not needed, as the analyzer does not flag ternary operators afaik. The other check always returns true for any if statement, which leads to the bug observed by @Yozer. This branch should also check the child nodes of the statement. To prevent the bug in my post above, it also has to check if the condition is negated.

https://github.com/dotnet/roslyn-analyzers/blob/45879f0b46557886b8d553d121a81866f67ce08a/src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Performance/BasicDoNotGuardDictionaryRemoveByContainsKey.vb#L17-L23

The first if should check for `SingleLineIfStatementSyntax` instead of `IfStatementSyntax`. Similarly to the C# case, both branches need to check if the condition is negated.

---

There were tests to check both cases, but they only asserted that there was a diagnostic reported, **not** that there was also no fix applied. See:

https://github.com/dotnet/roslyn-analyzers/blob/45879f0b46557886b8d553d121a81866f67ce08a/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs#L245-L261

and

https://github.com/dotnet/roslyn-analyzers/blob/45879f0b46557886b8d553d121a81866f67ce08a/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs#L167-L179

These should call `await VerifyCS.VerifyCodeFixAsync(source, source);` to check that there was no code fix applied. If you change this line, both tests fail as expected.

---

I can provide a PR for both of these issues. I will also check, if #6767 is affected as I based my test cases on this analyzer.
